caravel/models.py

```diff
@@ -284,6 +288,73 @@ def get_viz(self, url_params_multidict=None):
             slice_=self
         )
 
+    @classmethod
+    def import_slice(cls, slc, import_time=None):
+        """Inserts or overrides slc in the database.
+
+        remote_id and import_time fields in params_dict are set to track the
+        slice origin and ensure correct overrides for multiple imports.
+        Slice.perm is used to find the datasources and connect them.
+        """
+        session = db.session
+        make_transient(slc)
+        slc.dashboards = []
+        # perms are named [{db_name}].[{datasource_name}](id:{id})
+        id_pos = slc.perm.find('(id') if slc.perm else -1
+        if id_pos == -1:
+            logging.warning('slice {} has malformed perm {}'.format(
+                slc.slice_name, slc.perm))
+            return None
+
+        perm_prefix = slc.perm[:id_pos]
+        datasource_class = SourceRegistry.sources[slc.datasource_type]
+        datasources = session.query(datasource_class).all()
```

Comment:
filter by doesn't work for lookups by name, created function ` SourceRegistry.get_datasource_by_name`